File: backend/src/main/java/com/mulesoft/arc/arcdatastore/backend/models/ArcAnalyticsDailyResult.java
package com.mulesoft.arc.arcdatastore.backend.models;

/**
 * A class representing result for daily data query
 */

public abstract class ArcAnalyticsDailyResult {
    public final String kind;
    public int result;

    ArcAnalyticsDailyResult(String kind) {
        this.kind = kind;
    }
}

File: models/arc-analytics-daily-result.js
'use strict';

/**
 * A class representing result for daily data query.
 *
 * This class should be extended by class that returns specific results.
 */
class ArcAnalyticsDailyResult {

  constructor(kind, result) {
    /**
     * The default kind value. Should be overwritten by extending class.
     *
     * @type {String}
     */
    this.kind = kind || 'ArcAnalytics#UnknownResult';
    /**
     * Result of computation.
     *
     * @type {Number}
     */
    this.result = result || 0;
  }
}

module.exports.ArcAnalyticsDailyResult = ArcAnalyticsDailyResult;
